File: week 2/2.c
// 2. Write a C program to swap two numbers using pointers and functions ( Also implement this program in Python)


#include <stdio.h>

void swap(int *a, int *b)
{
    int temp;

    temp = *a;
    *a = *b;
    *b = temp;
}

int main()
{
    int x, y;

    printf("Enter two numbers: ");
    scanf("%d %d", &x, &y);

    printf("\nBefore Swapping:\n");
    printf("x = %d, y = %d\n", x, y);

    swap(&x, &y);

    printf("\nAfter Swapping:\n");
    printf("x = %d, y = %d\n", x, y);

    return 0;
}

File: week 2/2.py
# 2. Write a C program to swap two numbers using pointers and functions ( Also implement this program in Python)


def swap(a, b):
    return b, a

x = int(input("Enter first number: "))
y = int(input("Enter second number: "))

print("\nBefore Swapping:")
print("x =", x)
print("y =", y)

x, y = swap(x, y)

print("\nAfter Swapping:")
print("x =", x)
print("y =", y)

